src/main/java/com/github/piedpiper/node/stepfunctions/StepFunctionsBaseHandler.java

```diff
@@ -0,0 +1,35 @@
+package com.github.piedpiper.node.stepfunctions;
+
+import com.amazonaws.services.stepfunctions.AWSStepFunctions;
+import com.amazonaws.services.stepfunctions.AWSStepFunctionsClientBuilder;
+import com.github.piedpiper.aws.AccessKeySecretKeyCredentialProvider;
+import com.github.piedpiper.node.NodeInput;
+import com.github.piedpiper.node.NodeOutput;
+
+public abstract class StepFunctionsBaseHandler extends StepFunctionsBaseNode {
+	protected AWSStepFunctions stepFunctionsClient; 
+	private void setStepFunctionsClient(NodeInput input) throws Exception{
+		this.stepFunctionsClient = AWSStepFunctionsClientBuilder.standard()
+									.withRegion(getRegion(input))
+									.withCredentials(new AccessKeySecretKeyCredentialProvider(getAccessKey(input),getSecretKey(input)))
+									.build();
+		
+	}
+	
+	@Override
+	public NodeOutput apply(NodeInput input){
+		try{
+			setStepFunctionsClient(input);
```

Comment:
refere to DynamoDBBased note. its has a methoed called getDynamoDBClient(), Similarly you should have getStepFunctionClient (But this is also not according to bean conventions) You can use createdStepFunctionsClient